From 0a402fcb6f41be386bf2a9371991b502e34683c6 Mon Sep 17 00:00:00 2001
From: Shawn Guo <shawn.guo@freescale.com>
Date: Fri, 13 Jun 2014 11:37:29 +0800
Subject: [PATCH] ENGR00317981: drm: vivante: drm_platform_exit is gone
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The function drm_platform_exit() is removed by commit e2577d455adb (drm:
rip out drm_platform_exit).  To fix the build error below, we need to
call drm_put_dev directly.

  CC      drivers/gpu/drm/vivante/vivante_drv.o
drivers/gpu/drm/vivante/vivante_drv.c: In function ‘vivante_exit’:
drivers/gpu/drm/vivante/vivante_drv.c:100:3: error: implicit declaration of function ‘drm_platform_exit’ [-Werror=implicit-function-declaration]

Note, this only works with commit "ENGR00317981: drm: set drm_device
pointer into drvdata".

Signed-off-by: Shawn Guo <shawn.guo@freescale.com>
---
 drivers/gpu/drm/vivante/vivante_drv.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/vivante/vivante_drv.c b/drivers/gpu/drm/vivante/vivante_drv.c
index c9878e3..d5a915b 100644
--- a/drivers/gpu/drm/vivante/vivante_drv.c
+++ b/drivers/gpu/drm/vivante/vivante_drv.c
@@ -95,7 +95,8 @@ static int __init vivante_init(void)
 static void __exit vivante_exit(void)
 {
 	if (pplatformdev) {
-		drm_platform_exit(&driver, pplatformdev);
+		/* The drvdata is set in drm_get_platform_dev() */
+		drm_put_dev(platform_get_drvdata(pplatformdev));
 		platform_device_unregister(pplatformdev);
 		pplatformdev = NULL;
 	}
-- 
1.8.0

